index.js

```diff
@@ -92,9 +92,19 @@ function pluralTypeIndex(locale, count) {
   return pluralTypes[pluralTypeName(locale)](count);
 }
 
+function constructTokenRegex(opts) {
+  var escape = function (token) {
+    return token.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+  };
+
+  var prefix = (opts && opts.prefix) || '%{';
+  var suffix = (opts && opts.suffix) || '}';
+  return new RegExp(escape(prefix) + '(.*?)' + escape(suffix), 'g');
```

Comment:
I think this should either throw an exception if `prefix` and `suffix` are equal, or, we should have lots of tests with an identical prefix/suffix.

Reply:
I think we should allow `prefix` and `suffix` to be the same, it could be a valid use case IMO. So I decided to add some more tests instead.